File: packages/scalaModule/src/main/scala/application/child/Update.scala
package application.child
import base.AbstractUpdate

/**
  * Created by ryota on 2017/03/05.
  */

object Update extends AbstractUpdate[Msg, Model]  {
  def apply(msg: Msg, model: Model) = msg match {
    case Init => model
    case SetInteger(num) => model.copy(num = num)
  }
}

File: packages/kotlinModule/src/child/Update.kt
package child

/**
 * Created by ryota on 2017/03/05.
 */

fun update(model: child.Model, msg: Msg) : child.Model {
    return when (msg) {
        is child.Msg.Init -> model
        is child.Msg.Update -> model.copy(str = msg.str)
        else -> model
    }
}
